Token-wrap docs (#11)
solana-program/solana-program.github.io: docs/content/docs/token-wrap.mdx

        ```
    </Tab>

    <Tab value="JS">
        ```typescript
        import { address } from '@solana/kit';
        import {
            findBackpointerPda,
            findWrappedMintAuthorityPda,
            findWrappedMintPda,
        } from '@solana-program/token-wrap';
        import { TOKEN_2022_PROGRAM_ADDRESS } from '@solana-program/token-2022';

        const UNWRAPPED_MINT_ADDRESS = address('5StBUZ2w8ShDN9iF7NkGpDNNH2wv9jK7zhArmVRpwrCt');

        async function main() {
            const [wrappedMint] = await findWrappedMintPda({
                unwrappedMint: UNWRAPPED_MINT_ADDRESS,
                wrappedTokenProgram: TOKEN_2022_PROGRAM_ADDRESS,
            });
            const [backpointer] = await findBackpointerPda({ wrappedMint });
            const [wrappedMintAuthority] = await findWrappedMintAuthorityPda({ wrappedMint });

            console.log('WRAPPED_MINT_ADDRESS', wrappedMint);
            console.log('BACKPOINTER', backpointer);
            console.log('WRAPPED_MINT_AUTHORITY', wrappedMintAuthority);
        }

        void main();
        ```
    </Tab>
</Tabs>

### Create escrow account

> Before wrapping tokens, you need to create an account to hold the unwrapped tokens. It's mint authority must be the correct PDA (see `find-pdas` command above). There is also a helper to create this account:

Reviewer: nit typo and also maybe clearer?
```suggestion
Before wrapping tokens, you need to create an account to hold the unwrapped tokens. The escrow account's owner must be the correct PDA (see `find-pdas` command above). There is also a helper to create this account:
```

        $ WRAPPED_TOKEN_PROGRAM=TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb

        $ spl-token-wrap create-escrow-account $UNWRAPPED_MINT_ADDRESS $WRAPPED_TOKEN_PROGRAM

        Creating escrow account under program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA for unwrapped mint BVpjjYmSgSPZbFGTXe52NYXApsDNQJRe2qQF1hQft85e owned by PDA 8WdYPmtq8c6ZfmHMZUwCQL2E8qVHEV8rG9MXkyax3joR
        Escrow Account Address: 4NoeQJKuH8fu1Pqk5k8BJpNu4wA7T8K6QABJxjTWoHs3
        Escrow Account Owner (PDA): 8WdYPmtq8c6ZfmHMZUwCQL2E8qVHEV8rG9MXkyax3joR
        Escrow Token Program ID: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA
        Signature: 3ysN6YEQcsYQBjnCPMas9xEGP53CSSvoL6CSJ1vJS1S5ZvtN5NbsUtDKQMs6hwCQHhsctcrEhLQBLTEBuQWEKqNE
        ```
    </Tab>

    <Tab value="JS">
        ```typescript
        import {
            address,
            createKeyPairSignerFromBytes,
            createSolanaRpc,
            createSolanaRpcSubscriptions,
            sendAndConfirmTransactionFactory,
            signTransactionMessageWithSigners,
        } from '@solana/kit';
        import { TOKEN_2022_PROGRAM_ADDRESS } from '@solana-program/token-2022';

>         import { createEscrowAccountTx } from '../utilities';

Reviewer: nit: import from `@solana-program/token-wrap`

        Recipient wrapped token account: HKHfad5Rx7Vv1iWzPiQhx3cnXpbVfDonYRRo1e16x5Bt
        Escrow account: 4NoeQJKuH8fu1Pqk5k8BJpNu4wA7T8K6QABJxjTWoHs3
        Amount: 100
        Signers:
          26xTNzcurTuXQfHSCCuamxmrDXbkbA38JtGC9GhEcKgVZwxnyvXBD5AMH8TXmkfpNw64noDPaS4Ezm4RLMvfq3nF
        ```

        You can specify a recipient token account with the `--recipient-token-account` option. If not provided, the associated token account of the fee payer will be used or created if it doesn't exist.

        ```console
        $ spl-token-wrap wrap $UNWRAPPED_TOKEN_ACCOUNT $ESCROW_ACCOUNT $WRAPPED_TOKEN_PROGRAM 100 \
            --recipient-token-account $RECIPIENT_WRAPPED_TOKEN_ACCOUNT
        ```
    </Tab>

    <Tab value="JS">
        ```typescript
        import {
            address,
            createKeyPairSignerFromBytes,
            createSolanaRpc,
            createSolanaRpcSubscriptions,
            getSignatureFromTransaction,
            sendAndConfirmTransactionFactory,
            signTransactionMessageWithSigners,
        } from '@solana/kit';
        import { TOKEN_2022_PROGRAM_ADDRESS } from '@solana-program/token-2022';

>         import { singleSignerWrapTx } from '../wrap';

Reviewer: nit: import from `@solana-program/token-wrap`

            const { value: blockhash } = await rpc.getLatestBlockhash().send();

            const wrapMessage = await singleSignerWrapTx({
                rpc,
                blockhash,
                payer,
                unwrappedTokenAccount: UNWRAPPED_TOKEN_ACCOUNT,
                escrowAccount: ESCROW_ACCOUNT,
                wrappedTokenProgram: TOKEN_2022_PROGRAM_ADDRESS,
                recipientWrappedTokenAccount: RECIPIENT,
                amount: AMOUNT_TO_WRAP,
            });

            const signedWrapTx = await signTransactionMessageWithSigners(wrapMessage.tx);
            await sendAndConfirm(signedWrapTx, { commitment: 'confirmed' });
            const wrapSignature = getSignatureFromTransaction(signedWrapTx);

            console.log('======== Wrap Successful ========');
            console.log('Wrap amount:', wrapMessage.amount);
            console.log('Recipient account:', wrapMessage.recipientWrappedTokenAccount);
            console.log('Escrow Account:', wrapMessage.escrowAccount);
            console.log('Signature:', wrapSignature);
        }

        void main();
        ```
    </Tab>
</Tabs>

> ### Wrap tokens (multisig)
> 
> An example wrapping tokens whose origin is a token account owned by a multisig.

Reviewer: nit: To be totally clear that it's an SPL-Token multisig, and not a generic multisig
```suggestion
### Wrap tokens (SPL Token Multisig)

An example wrapping tokens whose origin is a token account owned by an SPL Token multisig.
```

<Tabs groupId="language" items={['CLI', 'JS']}>
    <Tab value="CLI">
        There are two parts to this. The first is having the multisig members sign the message independently.

>         The second is the broadcaster collecting those signatures and making a final call.

Reviewer: nit to be clearer
```suggestion
        The second is the broadcaster collecting those signatures and sending the transaction to the network.
```

        Recipient wrapped token account: HKHfad5Rx7Vv1iWzPiQhx3cnXpbVfDonYRRo1e16x5Bt
        Escrow account: 4NoeQJKuH8fu1Pqk5k8BJpNu4wA7T8K6QABJxjTWoHs3
        Amount: 23
        Signers:
          5pBReBRzy8yWLbz5j5GNBVFTmwGy65d4BvzigUPRTWWRYx4SUceNWDb78h1ufaRdzyi7yNmKpdLHv2eNS7ziaH7L
          3KdzhMYjFxBFZtQzEqfeugg6LVGhApSRj8pAx8HpgqCRU7C3gA2Wm5Hvx55taMAcpDWaKSJdtpgUJ8ksBVo4PDJU
          259kbWfYYhhe4FjTWZCeCXhPN4q3VSKN4dRHMcb42i85jptfh82TocrEf13aj5qMMDux9btzL5RCV55AxCWJbu5Q
        ```

        Note all three needed signers in final broadcasted message.
    </Tab>

    <Tab value="JS">
        ```typescript
        import {
            address,
            createKeyPairSignerFromBytes,
            createNoopSigner,
            createSolanaRpc,
            createSolanaRpcSubscriptions,
            getBase58Decoder,
            partiallySignTransactionMessageWithSigners,
            sendAndConfirmTransactionFactory,
        } from '@solana/kit';
        import { TOKEN_2022_PROGRAM_ADDRESS } from '@solana-program/token-2022';
        import { findWrappedMintAuthorityPda } from '../generated';
        import { multisigOfflineSignWrapTx } from '../wrap';

>         import { combinedMultisigTx, getOwnerFromAccount } from '../utilities';

Reviewer: I'm guessing all of these can come from the package too

        ```console
        $ ESCROW_ACCOUNT=4NoeQJKuH8fu1Pqk5k8BJpNu4wA7T8K6QABJxjTWoHs3
        $ WRAPPED_TOKEN_ACCOUNT=HKHfad5Rx7Vv1iWzPiQhx3cnXpbVfDonYRRo1e16x5Bt
        $ UNWRAPPED_TOKEN_RECIPIENT=DKFjYKEFS4tkXjamwkuiGf555Lww3eRSWwNTbue9x14

        $ spl-token-wrap unwrap $WRAPPED_TOKEN_ACCOUNT $ESCROW_ACCOUNT $UNWRAPPED_TOKEN_RECIPIENT 50

        Unwrapping 50 tokens from mint BVpjjYmSgSPZbFGTXe52NYXApsDNQJRe2qQF1hQft85e
        Unwrapped token program: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA
        Unwrapped mint address: BVpjjYmSgSPZbFGTXe52NYXApsDNQJRe2qQF1hQft85e
        Recipient unwrapped token account: DKFjYKEFS4tkXjamwkuiGf555Lww3eRSWwNTbue9x14
        Amount unwrapped: 50
        Signers:
          4HjHkjpjZztvoYT95mKHy2wH7z7iAFqpxSMeeMdUpPTzsjZN3vKg1KXvTMV7VT3jK6CaePYYXYDCTm52KTWz6du
        ```
    </Tab>

    <Tab value="JS">
        ```typescript
        import {
            address,
            createKeyPairSignerFromBytes,
            createSolanaRpc,
            createSolanaRpcSubscriptions,
            getSignatureFromTransaction,
            sendAndConfirmTransactionFactory,
            signTransactionMessageWithSigners,
        } from '@solana/kit';

>         import { singleSignerUnwrapTx } from '../unwrap';

Reviewer: nit: import from `@solana-program/token-wrap`

            const sendAndConfirm = sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions });

            const payer = await createKeyPairSignerFromBytes(PRIVATE_KEY_PAIR);
            const { value: blockhash } = await rpc.getLatestBlockhash().send();

            const unwrapMessage = await singleSignerUnwrapTx({
                rpc,
                blockhash,
                payer,
                wrappedTokenAccount: WRAPPED_TOKEN_ACCOUNT,
                unwrappedEscrow: ESCROW,
                amount: AMOUNT_TO_WRAP,
                recipientUnwrappedToken: RECIPIENT,
            });

            const signedUnwrapTx = await signTransactionMessageWithSigners(unwrapMessage.tx);
            await sendAndConfirm(signedUnwrapTx, { commitment: 'confirmed' });
            const unwrapSignature = getSignatureFromTransaction(signedUnwrapTx);

            console.log('======== Unwrap Successful ========');
            console.log('Unwrapped amount:', unwrapMessage.amount);
            console.log('Recipient account:', unwrapMessage.recipientUnwrappedToken);
            console.log('Signature:', unwrapSignature);
        }

        void main();
        ```
    </Tab>
</Tabs>

> ### Unwrap tokens (multisig)
> 
> An example unwrapping tokens whose origin is a token account owned by a multisig.

Reviewer: nit: same here for clarity
```suggestion
### Unwrap tokens (SPL Token Multisig)

An example unwrapping tokens whose origin is a token account owned by an SPL Token multisig.
```

<Tabs groupId="language" items={['CLI', 'JS']}>
    <Tab value="CLI">
        There are two parts to this. The first is having the multisig members sign the message independently.

>         The second is the broadcaster collecting those signatures and making a final call.

Reviewer: nit: as before
```suggestion
        The second is the broadcaster collecting those signatures and sending the transaction to the network.
```

        Recipient unwrapped token account: DKFjYKEFS4tkXjamwkuiGf555Lww3eRSWwNTbue9x14
        Amount unwrapped: 5
        Signers:
          5s2gNCExGchZJcnDTHMubyRsjuNprhyzaNeSbc34sJSLt5AB8N6V6uBoAegnBF1zvm1s65CPtjVLNH7Eb2hFYLsM
          eEcQFqdsDZzKL5CAm43kofVvnXCy7ZTQFm3RS6iYg4nckrSUzWZctebDYqcUNqtxxBgnLHyeDZiYMxNABAYYt2x
          5fxWE9KQYFQHk2u9ienicDtuaRf9XWBvrmM48CBTxwtmpJXuxHxDzAYSM5atHe77rFTVsezbLCbuzirN1o5XdZTf
        ```

        Note all three needed signers in final broadcasted message.
    </Tab>

    <Tab value="JS">
        ```typescript
        import {
            address,
            createKeyPairSignerFromBytes,
            createNoopSigner,
            createSolanaRpc,
            createSolanaRpcSubscriptions,
            getBase58Decoder,
            partiallySignTransactionMessageWithSigners,
            sendAndConfirmTransactionFactory,
        } from '@solana/kit';
        import { TOKEN_2022_PROGRAM_ADDRESS } from '@solana-program/token-2022';
        import { findWrappedMintAuthorityPda } from '../generated';
        import { combinedMultisigTx, getOwnerFromAccount } from '../utilities';

>         import { multisigOfflineSignUnwrap } from '../unwrap';

Reviewer: nit: import from `@solana-program/token-wrap`
